### DISCUSSION TITLE
use cython to speed up geopack

### DISCUSSION THREAD

**Discussion by Keoeeit** [spedas/pyspedas#620] [Ideas]
Since there is no vectorization, the efficiency of the python version of geopack is very low. Is it possible to consider using cython to directly call the fortran version of geopack to speed up?

**Comment by jameswilburlewis** [comment]
Sorry, just now seeing this!  Yes, it might be beneficial to implement the models as Fortran (or C) libraries which could be distributed as binary wheels.   Porting the code to C might make for a better end user experience, if a binary wheel is not yet available for their platform and they need to compile the library locally.   

Ideally, we would get a geopack build recipe set up under conda, which might facilitate the process of compiling for multiple platforms.  This is also a prerequisite for making pyspedas itself available via conda, which we definitely want to work toward.

**Comment by Keoeeit** [comment]
Looking forward to performance improvements, thanks very much for the reply
